lib/workers/repository/process/extract-update.ts

```diff
@@ -68,9 +68,22 @@ function extractStats(
 export function isCacheExtractValid(
   baseBranchSha: string,
   configHash: string,
-  cachedExtract?: BaseBranchCache
+  cachedExtract?: BaseBranchCache,
+  isOnboarded?: boolean,
+  onboardingBranch?: string
```

Comment:
I prefer not to extend it like this. Instead, find an earlier location to invalid all extract caches if the onboarding SHA has changed since previous run

Reply:
We can check for `isBranchModified` here and remove the extract-cache if true.
https://github.com/RahulGautamSingh/renovate/blob/05ddfaccd7f0a551c74de06f0f99e1fe19ceae9b/lib/workers/repository/onboarding/branch/index.ts#L44-L48

Reply:
`isBranchModified` internally checks the cache too. But we don't currently store `onaboardingBranch` info in the `branchCache`. I will add that functionality here: https://github.com/renovatebot/renovate/pull/20886

Reply:
Too many changes for one PR I think as we will need to handle creating, removing and syncing cache in this PR.